File: cobalt/backend/application/dtos/metrics.py
#  Copyright (C) 2026 ArtoriasCode
#  Author: ArtoriasCode
#  Repository: https://github.com/ArtoriasCode/cobalt
#  SPDX-License-Identifier: AGPL-3.0-or-later

from datetime import datetime

from application.dtos.base import BaseDto


class MetricDto(BaseDto):
    value: float
    date: datetime

class MetricDiskDto(BaseDto):
    free: int
    total: int
    last_check: datetime
    next_check: datetime

class MetricsSubscribeServerDto(BaseDto):
    server_id: int

class MetricsUnsubscribeServerDto(BaseDto):
    server_id: int

File: cobalt/frontend/src/types/entities/metrics.ts
/*
 * Copyright (C) 2026 ArtoriasCode
 * Author: ArtoriasCode
 * Repository: https://github.com/ArtoriasCode/cobalt
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */

/**
 * Represents a single CPU or RAM metric data point.
 */
export interface MetricEntity {
  value: number
  date: string
}

/**
 * Represents host disk usage metrics.
 */
export interface DiskMetricsEntity {
  free: number
  total: number
  last_check: string
  next_check: string
}